packages/langium/src/documents/document-builder.ts

```diff
@@ -103,7 +104,7 @@ export class DefaultDocumentBuilder implements DocumentBuilder {
         document.precomputedScopes = this.scopeComputation.computeScope(document);
     }
 
-    protected isAffected(document: LangiumDocument, changedUri: string): boolean {
-        return changedUri !== document.textDocument.uri;
+    protected isAffected(document: LangiumDocument, changedUri: URI): boolean {
+        return changedUri.toString() !== document.uri.toString();
```

Comment:
Istn't there a way to compare the URI objects?

Reply:
There is none. However, we could write our own utility functions for that. I'm also wondering, whether we should re-export the URI in langium, as that will be used a fair bit by downstream users.

Reply:
A utility function might be useful in case we need to normalize the URI somehow, but let's keep that in mind for later.

Re-exporting the URI makes sense because it's an important part of our API.